api/api.py

```diff
@@ -0,0 +1,188 @@
+        super(Button, self).__init__()
+
+    def post(self):
+        args = self.parser.parse_args()
+        button = request.base_url.split("/")[-1]
+        if button == "both":
+            seph.handle_button(1, True)
+            seph.handle_button(2, True)
+            time.sleep(args.delay)
+            seph.handle_button(1, False)
+            seph.handle_button(2, False)
+        else:
+            buttons = { "left": 1, "right": 2 }
+            actions = { "press": True, "release": False }
+
+            assert button in buttons.keys()
+            action = args.get("action", None)
+
+            if action is None:
+                seph.handle_button(buttons[button], True)
+                time.sleep(args.delay)
+                seph.handle_button(buttons[button], False)
+            else:
+                seph.handle_button(buttons[button], actions[action])
+
+        return {}, 200
+
+class EventClient:
+    def __init__(self):
+        self.events = []
+
+    def generate(self):
+        try:
+            # force headers to be sent
+            yield b""
+
+            while True:
+                with events.condition:
+                    events.condition.wait(1)
+
+                while self.events:
+                    event = self.events.pop(0)
+                    data = json.dumps(event).encode()
+                    yield data + b"\n"
+        finally:
+            events.remove_client(self)
+
+    def send_screen_event(self, event):
+        self.events.append(event)
+
+class Events(Resource):
+    def __init__(self):
+        self.parser = reqparse.RequestParser()
+        self.parser.add_argument("stream", type=inputs.boolean, default=True)
+        super(Events, self).__init__()
+
+    def get(self):
+        args = self.parser.parse_args()
+        if args.stream:
+            client = EventClient()
+            events.add_client(client)
+            return Response(stream_with_context(client.generate()), content_type="text/event-stream")
+        else:
+            return {"events": events.events}, 200
+
+    def delete(self):
+        events.events.clear()
+        return {}, 200
+
+class Finger(Resource):
+    def __init__(self):
+        self.parser = reqparse.RequestParser()
+        self.parser.add_argument("x", type=int, required=True)
+        self.parser.add_argument("y", type=int, required=True)
+        self.parser.add_argument("action", choices=["press", "release"])
+        self.parser.add_argument("delay", type=float, default=0.1)
+        super(Finger, self).__init__()
+
+    def post(self):
+        args = self.parser.parse_args()
+        x = args.get("x")
+        y = args.get("y")
+        action = args.get("action", None)
+
+        if action is None:
+            seph.handle_finger(x, y, True)
+            time.sleep(args.delay)
+            seph.handle_finger(x, y, False)
+        else:
+            seph.handle_finger(x, y, actions[action])
```

Comment:
`actions` is not defined here, so calling this API leads to:

    NameError: name 'actions' is not defined

Reply:
done